openedx/features/philu_courseware/views.py

```diff
@@ -65,3 +65,23 @@ def record_and_fetch_competency_assessment(request, chapter_id):
             }, status=status.HTTP_201_CREATED)
     else:
         return Response({'message': serializer.errors}, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(['POST'])
+@view_auth_classes(is_authenticated=True)
+@renderer_classes([JSONRenderer, BrowsableAPIRenderer])
+def revert_user_post_assessment_attempts(request, course_id):
+    user = request.user
+    problem_id = request.data.get('problem_id')
+    validated_problem_id = validate_problem_id(problem_id)
+    reverted_attempts_count = CompetencyAssessmentRecord.objects.revert_user_post_assessment_attempts(
+                                problem_id=validated_problem_id, user=user)
+    has_deleted_post_assessment_attempts = reverted_attempts_count > 0
+    if has_deleted_post_assessment_attempts:
+        revert_user_attempts_from_edx(course_id, user, validated_problem_id)
+        success_response = {
+            'message': _('Your attempts has been reverted successfully')
+        }
+        return Response(success_response, status=status.HTTP_200_OK)
+    else:
+        raise ValidationError(_('This user has no attempts for this problem id in post assessment'))
```

Comment:
Shouldn't we just `return Response(success_response, status=status.HTTP_200_OK)` here as well? Why would we want to expose more information in an API endpoint than we need to? This, of course, depends on if we trust on the correctness of records saved in the `CompetencyAssessmentRecords`.

Reply:
Actually on frontend we ll be reloading the page if it reverted attempts. But if it didn't revert any attempt then that reloading can be saved on this failure or may be we can send success response with more information like sending number of records reverted? 
WDYT?

Reply:
We will not be showing the reset assessment button, if the assessment has not been attempted. Please see point 1, subpoint 4 in AC of [LP-2162](https://philanthropyu.atlassian.net/browse/LP-2162). Does that make sense, or am I missing something?